LibSWBF2/Chunks/LVL/lght/lght.cpp

```diff
@@ -15,33 +18,50 @@ namespace LibSWBF2::Chunks::LVL::light
 		THROW("Not implemented!");
 	}
 
+    /*
+     There isn't much abstraction here, just basic exposure
+     of the chunk structure.  All light semantics are parsed from
+     the SCOP/DATA chunks in the Light wrapper classes.
+     */
+
 	void lght::ReadFromStream(FileReader& stream)
-	{
-		THROW("Not implemented!");
+	{       
+        BaseChunk::ReadFromStream(stream);
+        Check(stream);
+        
+        /*
+         It seems in every ucfb, the first lght chunk contains
+         all local lights, and the other per-world lght's merely index
+         into the first one.  So for now, before we do per-world stuff,
+         we just parse this first chunk.
+         */
 
-		/*
-		BaseChunk::ReadFromStream(stream);
-		Check(stream);
+        if (!lght::skip)
+        {
+            STR<"NAME"_m> *p_Marker;
+            READ_CHILD(stream, p_Marker);
+            delete p_Marker;
 
-		READ_CHILD(stream, p_Name);
-		READ_CHILD(stream, p_Info);
+            //PositionInChunk might not be best for this
+            while (BaseChunk::PositionInChunk(stream.GetPosition()))
+            {
+                DATA *tempHeader;
+                SCOP *tempScope;
+                
+                READ_CHILD(stream, tempHeader);
+                READ_CHILD(stream, tempScope);
 
-		while (ThereIsAnother(stream))
-		{
-			FMT_* fmt;
-			READ_CHILD(stream, fmt);
-			m_FMTs.Add(fmt);
-		}
+                //This in combo with PositionInChunk might be a bad way...
+                ForwardToNextHeader(stream);
+            }
```

Comment:
I did this from intuition, perhaps there's a more conformant way?

Reply:
Yes, take a look for example at `LVL.modl.cpp`:

```
void modl::ReadFromStream(FileReader& stream)
{
    BaseChunk::ReadFromStream(stream);
    Check(stream);

    while (ThereIsAnother(stream))
    {
        ChunkHeader next = stream.ReadChunkHeader(true);
        if (next == "NAME"_h)
        {
            READ_CHILD(stream, p_Name);
        }
        else if (next == "NODE"_h)
        {
            READ_CHILD(stream, p_Node);
        }
        else if (next == "INFO"_h)
        {
            READ_CHILD(stream, p_Info);
        }
        else if (next == "segm"_h)
        {
            segm* segment;
            READ_CHILD(stream, segment);
            m_Segments.Add(segment);
        }
        else
        {
            READ_CHILD_GENERIC(stream);
        }
    }

    BaseChunk::EnsureEnd(stream);
}
```

Reply:
Done